### PR TITLE
[mlir][python] Enable python bindings for Index dialect

### PR DESCRIPTION
**Pull request by stevenvar**
This small patch enables python bindings for the index dialect.

### CODE REVIEW

File: mlir/python/mlir/dialects/IndexOps.td [hunk 1]
```diff
@@ -0,0 +1,14 @@
+//===-- IndexOps.td - Entry point for Index bindings -----*- tablegen -*-===//
```
Comment (ftynse) [inline mlir/python/mlir/dialects/IndexOps.td:1]: Nit: please pad to 80 columns.

File: mlir/test/python/dialects/index_dialect.py [hunk 2]
```diff
@@ -0,0 +1,335 @@
...
+
+
+# CHECK-LABEL: TEST: testConstantOp
+@run
+def testConstantOp():
+  with Context() as ctx, Location.unknown():
+    module = Module.create()
+    with InsertionPoint(module.body):
+      a = index.ConstantOp(value=42)
+    # CHECK: %[[A:.*]] = index.constant 42
```
Comment (ftynse) [inline mlir/test/python/dialects/index_dialect.py:19]: Nit: there is no point it capturing the result of a regular expression if it is not being used, prefer `%{{.*}}` instead.

File: mlir/test/python/dialects/index_dialect.py [hunk 3]
```diff
@@ -0,0 +1,335 @@
...
+
+# CHECK-LABEL: TEST: testAndOp
+@run
+def testAndOp():
+  with Context() as ctx, Location.unknown():
+    module = Module.create()
+    with InsertionPoint(module.body):
+      a = index.ConstantOp(value=42)
+      r = index.AndOp(a, a)
+    # CHECK: %[[R:.*]] = index.and {{.*}}, {{.*}}
```
Comment (ftynse) [inline mlir/test/python/dialects/index_dialect.py:42]: Nit: prefer the `%{{.*}}` form as it makes tests slightly more understandable.

File: mlir/test/python/dialects/index_dialect.py [hunk 4]
```diff
@@ -0,0 +1,335 @@
...
+    print(module)
+
+
+# CHECK-LABEL: TEST: testOrOp
+@run
+def testOrOp():
+  with Context() as ctx, Location.unknown():
+    module = Module.create()
+    with InsertionPoint(module.body):
```
Comment (ftynse) [inline mlir/test/python/dialects/index_dialect.py:51]: Could we lift all this module creation and printing boilerplate into the `@run` decorator?

Comment (stevenvar) [reply mlir/test/python/dialects/index_dialect.py]: Surely! Thank you for the suggestion

Comment (stevenvar) [reply mlir/test/python/dialects/index_dialect.py]: > Could we lift all this module creation and printing boilerplate into the `@run` decorator?

Surely! Thank you for the suggestion

**Review by ftynse — APPROVED** [review]

**Pull request merged**